utils/system-notification/files/usr/libexec/rpcd/system.notification

```diff
@@ -0,0 +1,190 @@
+#!/bin/sh
+#
+# Copyright (C) 2021 TDT AG <development@tdt.de>
+#
+# This is free software, licensed under the Apache License Version 2.0.
+# See https://www.apache.org/licenses/LICENSE-2.0 for more information.
+#
+
+. /usr/share/libubox/jshn.sh
+
+MESSAGE_DIR="/tmp/notification/messages.d"
+
+get_uptime() {
+	local uptime
+
+	uptime=$(cat /proc/uptime)
+	echo "${uptime%%.*}"
+}
+
+get_id() {
+	local file="$1"
+	local id
+
+	id="${file##*/}"
+
+	echo "${id%*.json}"
+}
+
+list() {
+	json_init
+
+	json_add_object "add"
+	json_add_string "service"
+	json_add_string "title"
+	json_add_string "message"
+	json_close_object
+
+	json_add_object "remove"
+	json_add_int "number"
+	json_close_object
+
+	json_add_object "list"
+	json_close_object
+
+	json_add_object "flush"
+	json_close_object
+
+	json_dump
+}
+
+notification_add() {
+	local service="$1"
+	local title="$2"
+	local message="$3"
+
+	local timestamp uptime
+
+	local code=-1
+	local msg="Unable to add Notification"
+
+	timestamp="$(date +%s)"
+	uptime="$(get_uptime)"
+
+	json_init
+	json_add_string service "$service"
+	json_add_string title "$title"
+	json_add_string message "$message"
+	json_add_int timestamp "$timestamp"
+	json_dump > "${MESSAGE_DIR}/${uptime}.json"
+
+	if [ -f "${MESSAGE_DIR}/${uptime}.json" ]; then
+		code=0
+		msg="Notification added"
+	fi
+
+	json_init
+	json_add_int "code" "$code"
+	json_add_string "message" "$msg"
+	json_dump
+}
+
+notification_remove() {
+	local index="$1"
+	local code=-1
+	local msg="Notification not found"
+
+	local id
+
+	for entry in "$MESSAGE_DIR"/*.json; do
+		id="$(get_id "$entry")"
```

Comment:
is there anythign wrong with just using "basename" ?

Reply:
thank you I will change this

Reply:
@karlp I have now taken a closer look.
I need not only the name of the file here but also the ID.
I have to return this via ubus so user can use the ID to delete a message with the delete command